src/main/java/com/yahoo/bullet/parsing/BinaryExpression.java

```diff
@@ -39,6 +40,18 @@ public BinaryExpression() {
         right = null;
     }
 
+    @Override
+    public List<String> getRequiredFields() {
+        List<String> result = new ArrayList<>(left.getRequiredFields());
+        List<String> rightRequired = right.getRequiredFields();
+        for (String s : rightRequired) {
```

Comment:
You could potentially just use Set<String>

Reply:
Done.